test/reverseregistrar_test.js

```diff
@@ -28,47 +31,47 @@ describe('ReverseRegistrar', function() {
         registrarCode = utils.compileContract(['interface.sol', 'ReverseRegistrar.sol']).contracts['ReverseRegistrar.sol:ReverseRegistrar'];
     });
 
-    beforeEach(function(done) {
-        async.series([
-            function(done) { ens = utils.deployENS(accounts[0], done); },
-            function(done) {
-                registrar = web3.eth.contract(JSON.parse(registrarCode.interface)).new(
-                    ens.address,
-                    0,
-                    {
-                        from: accounts[0],
-                        data: registrarCode.bytecode,
-                        gas: 4700000
-                    }, function(err, contract) {
-                        if(contract.address != undefined)
-                            ens.setOwner(0, registrar.address, {from: accounts[0]}, done);
-                    });
-            }],
-            done
-        );
+    beforeEach(function() {
+        this.timeout(4000);
+        return utils.deployENSAsync(accounts[0])
+            .then(function(_ens) {
+                ens = _ens;
+                return new Promise(function(resolve, reject) {
+                    web3.eth.contract(JSON.parse(registrarCode.interface)).new(
```

Comment:
The "right" value to resolve is the second call, because the first has no address yet. So the `promisifyAll` does not work.
On the other hand, we could make a generic promisifier that passes on the second call only.